### PR TITLE
Document printExprWithStats debugging tool

### PR DESCRIPTION
**Pull request by mbasmanova**

### CODE REVIEW

File: velox/docs/develop/debugging/print-expr-with-stats.rst [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+    mod [cpu time: 50.49us, rows: 1024] -> BIGINT [#1]
+       cast(plus as BIGINT) [cpu time: 68.15us, rows: 1024] -> BIGINT [#2]
+          plus [cpu time: 51.84us, rows: 1024] -> INTEGER [#3]
+             c0 [cpu time: 0ns, rows: 0] -> INTEGER [#4]
+             c1 [cpu time: 0ns, rows: 0] -> INTEGER [#5]
+       5:BIGINT [cpu time: 0ns, rows: 0] -> BIGINT [#6]
+
+    mod [cpu time: 49.29us, rows: 1024] -> BIGINT [#7]
+       cast((plus(c0, c1)) as BIGINT) -> BIGINT [CSE #2]
```
Comment (aditi-pandit) [inline velox/docs/develop/debugging/print-expr-with-stats.rst:69]: Nit : Would be nice to highlight the usage of "CSE" if possible.

File: velox/docs/develop/debugging/print-expr-with-stats.rst [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+       3:BIGINT [cpu time: 0ns, rows: 0] -> BIGINT [#8]
+
+Another example is dictionary memoization. If we evaluate an expression over a
+column that uses the same base dictionary, the evaluation remembers the results
```
Comment (aditi-pandit) [inline velox/docs/develop/debugging/print-expr-with-stats.rst:73]: Nit : Make explicit the expression evaluation happens over "multiple batches of data of the column using the same base dictionary". When reading I felt I got this context of multiple batches from the later sentences.

File: velox/docs/develop/debugging/print-expr-with-stats.rst [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+.. code-block:: c++
+
+    /// Listener invoked on ExprSet destruction.
+    class ExprSetListener {
+     public:
+      virtual ~ExprSetListener() = default;
+
+      /// Called on ExprSet destruction. Provides runtime statistics about
+      /// expression evaluation.
+      /// @param uuid Universally unique identifier of the set of expressions.
```
Comment (aditi-pandit) [inline velox/docs/develop/debugging/print-expr-with-stats.rst:138]: How is the uuid generated ? For a given example with multiple expression sets how can we get the association between the expression and the uuid ?

**Review by aditi-pandit — COMMENTED** [review]
Thanks Masha for this explanation. It is very useful.

File: velox/docs/develop/debugging/print-expr-with-stats.rst (hunk 1 above)
Comment (mbasmanova) [reply velox/docs/develop/debugging/print-expr-with-stats.rst:69]: That would be nice, but I don't think I know how to do that.

File: velox/docs/develop/debugging/print-expr-with-stats.rst (hunk 3 above)
Comment (mbasmanova) [reply velox/docs/develop/debugging/print-expr-with-stats.rst:138]: UUID is auto-generated for each instance of ExprSet, which represents one or more expressions.

```
  ExprSet(
      std::vector<std::shared_ptr<const core::ITypedExpr>>&& source,
      core::ExecCtx* execCtx,
      bool enableConstantFolding = true);
...

ExprSet::~ExprSet() {
  listeners().withRLock([&](auto& listeners) {
    if (!listeners.empty()) {
      std::unordered_map<std::string, exec::ExprStats> stats;
      for (const auto& expr : exprs()) {
        addStats(*expr, stats);
      }

      auto uuid = makeUuid();
      for (auto& listener : listeners) {
        listener->onCompletion(uuid, {stats});
      }
    }
  });
}
```

File: velox/docs/develop/debugging/print-expr-with-stats.rst (hunk 2 above)
Comment (mbasmanova) [reply velox/docs/develop/debugging/print-expr-with-stats.rst]: Good suggestion. Updated.

**Review by aditi-pandit — APPROVED** [review]
Thanks !

**Pull request closed without merging**